segmentation fault on linux x86_64

i got this


  Compiling php-sys v0.1.0 (file:///root/php-rpm/php-sys)
   Compiling php-rpm v0.1.0 (file:///root/php-rpm)
error[E0425]: cannot find value `sapi_globals` in module `ffi`
  --> src/sapi.rs:76:14
   |
76 |         ffi::sapi_globals.sapi_headers.http_response_code = 200;
   |              ^^^^^^^^^^^^ did you mean `sapi_globals_id`?

error[E0425]: cannot find value `sapi_globals` in module `ffi`
  --> src/sapi.rs:79:14
   |
79 |         ffi::sapi_globals.server_context = context_ptr as *mut c_void;
   |              ^^^^^^^^^^^^ did you mean `sapi_globals_id`?

error[E0425]: cannot find value `sapi_globals` in module `ffi`
  --> src/sapi.rs:84:28
   |
84 |         let context = ffi::sapi_globals.server_context as *mut ServerContext;
   |                            ^^^^^^^^^^^^ did you mean `sapi_globals_id`?

error[E0425]: cannot find value `sapi_globals` in module `ffi`
  --> src/sapi.rs:90:37
   |
90 |         drop(CString::from_raw(ffi::sapi_globals.request_info.cookie_data));
   |                                     ^^^^^^^^^^^^ did you mean `sapi_globals_id`?

error[E0425]: cannot find value `sapi_globals` in module `ffi`
  --> src/sapi.rs:91:14
   |
91 |         ffi::sapi_globals.server_context = ptr::null_mut();
   |              ^^^^^^^^^^^^ did you mean `sapi_globals_id`?

error[E0425]: cannot find value `sapi_globals` in module `ffi`
   --> src/sapi.rs:143:10
    |
143 |     ffi::sapi_globals.request_info = request_info;
    |          ^^^^^^^^^^^^ did you mean `sapi_globals_id`?

error[E0425]: cannot find value `sapi_globals` in module `ffi`
   --> src/sapi.rs:163:24
    |
163 |     let context = ffi::sapi_globals.server_context as *mut ServerContext;
    |                        ^^^^^^^^^^^^ did you mean `sapi_globals_id`?

error[E0425]: cannot find value `sapi_globals` in module `ffi`
   --> src/sapi.rs:171:13
    |
171 |     if ffi::sapi_globals.headers_sent == 1 {
    |             ^^^^^^^^^^^^ did you mean `sapi_globals_id`?

error[E0425]: cannot find value `sapi_globals` in module `ffi`
   --> src/sapi.rs:197:24
    |
197 |     let context = ffi::sapi_globals.server_context as *mut ServerContext;
    |                        ^^^^^^^^^^^^ did you mean `sapi_globals_id`?

error[E0425]: cannot find value `sapi_globals` in module `ffi`
   --> src/sapi.rs:227:10
    |
227 |     ffi::sapi_globals.headers_sent = 1;
    |          ^^^^^^^^^^^^ did you mean `sapi_globals_id`?

error[E0425]: cannot find value `sapi_globals` in module `ffi`
   --> src/sapi.rs:236:24
    |
236 |     let context = ffi::sapi_globals.server_context as *mut ServerContext;
    |                        ^^^^^^^^^^^^ did you mean `sapi_globals_id`?

error[E0425]: cannot find value `sapi_globals` in module `ffi`
   --> src/sapi.rs:263:24
    |
263 |     let context = ffi::sapi_globals.server_context as *mut ServerContext;
    |                        ^^^^^^^^^^^^ did you mean `sapi_globals_id`?

error[E0425]: cannot find value `sapi_globals` in module `ffi`
   --> src/sapi.rs:325:24
    |
325 |     let context = ffi::sapi_globals.server_context as *mut ServerContext;
    |                        ^^^^^^^^^^^^ did you mean `sapi_globals_id`?

error: aborting due to 13 previous errors

error: Could not compile `php-rpm`.

To learn more, run the command again with --verbose.

--- a/src/sapi.rs
+++ b/src/sapi.rs
@@ -83,7 +83,9 @@ pub fn execute(method: hyper::Method, uri: hyper::Uri, http_version: hyper::Http
 
         // TODO move this into the request shutdown block
         // note: strangely enough, php_request_shutdown will not call our request shutdown callback
-        drop(CString::from_raw(php::sapi_globals.request_info.cookie_data));
+        if !php::sapi_globals.request_info.cookie_data.is_null() {
+            drop(CString::from_raw(php::sapi_globals.request_info.cookie_data));
+        }
         php::sapi_globals.request_info.cookie_data = ptr::null_mut();
         php::sapi_globals.server_context = ptr::null_mut();
 

